workspaces/src/network/testnet.rs

```diff
@@ -67,15 +68,25 @@ impl std::fmt::Debug for Testnet {
 impl AllowDevAccountCreation for Testnet {}
 
 #[async_trait]
-impl TopLevelAccountCreator for Testnet {
-    async fn create_tla(
+impl DevAccountCreator for Testnet {
+    async fn create_dev_account(
         &self,
         worker: Worker<dyn Network>,
         id: AccountId,
         sk: SecretKey,
         // TODO: return Account only, but then you don't get metadata info for it...
     ) -> Result<Execution<Account>> {
         let url = Url::parse(HELPER_URL).unwrap();
+        //only registrar can create tla on testnet, so must concatenate random created id with .testnet
+        let id = if self.info().name.eq("testnet") {
+            AccountId::from_str(format!("{}.testnet", id.as_str()).as_str())
```

Comment:
Let's make the `id` argument (`create_dev_account`/`create_sponsored_account`'s function argument) explicitly named as  `subaccount_prefix` or something along those lines, and document that on the trait - it is expected that this name does not contain `.` (`testnet` cannot create `sub.sub.testnet` accounts).